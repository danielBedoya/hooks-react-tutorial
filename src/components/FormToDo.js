import React from 'react'
import { useForm } from "../hooks/useForm";

export const FormToDo = ({dispatch}) => {

    const [{ description }, onHandleChange, reset] = useForm({
      description: "",
    });

    const addTodo = (e) => {
      e.preventDefault();

      if (description.trim().length <= 1) {
        return;
      }

      const action = {
        type: "add",
        payload: {
          id: new Date().getTime(),
          desc: description,
          done: false,
        },
      };

      dispatch(action);
      reset();
    };

    return (
        <div>
            <h5>Agregar tarea</h5>
            <form className="d-grid gap-2" onSubmit={ addTodo }>
                <input
                    type="text"
                    className="form-control"
                    name="description"
                    value={ description }
                    onChange={ onHandleChange }
                    >
                </input>
                <button className="btn btn-outline-primary mt-2" >Agregar</button>
            </form>
        </div>
    )
}
