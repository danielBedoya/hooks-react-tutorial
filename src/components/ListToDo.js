import React from 'react'

export const ListToDo = ({todos, dispatch}) => {

    const delTodo = (id) => {
      
      const action = {
        type: "delete",
        payload: {
          id: id
        }
      }

      dispatch(action);

    }

    const handleToggle = (id) => {
      const action = {
        type: "toggle",
        payload: {
          id: id,
        },
      };

      dispatch(action);
    };

    return (
        <ul style={{ display: "inline-block" }}>
            {todos.map((todo, i) => (
            <li key={todo.id}>
                <h5 
                    className={`card-title ${ todo.done && 'complete'}`}
                    style={{ display: "inline-block" }}
                    onClick={() => handleToggle(todo.id)}
                >
                    {i + 1}. {todo.desc}
                </h5>
                <div 
                    className="btn btn-outline-danger m-2" 
                    onClick={() => delTodo(todo.id)}
                >
                    Borrar
                </div>
            </li>
            ))}
        </ul>
    )
}
