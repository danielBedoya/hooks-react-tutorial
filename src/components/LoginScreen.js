import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const {user, setUser} = useContext(UserContext)

    const newUser = {
        id: user.id + 1,
        name: "Daniel"
    }

    return (
        <div>
            <h1>Login</h1>
            <div className="btn btn-primary"
                onClick={()=>setUser(newUser)}
            >
                Cambiar Usuario
            </div>
        </div>
    )
}
