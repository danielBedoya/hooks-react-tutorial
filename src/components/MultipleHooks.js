import React from 'react'
import { useFetch } from "../hooks/useFetch";
import { useCounter } from "../hooks/useCounter";

export const MultipleHooks = () => {

    let { counter, increment } = useCounter(1);
    let { loading, data, error } = useFetch(`https://breakingbadapi.com/api/quotes/${counter}`);
    let { author, quote} = !!data && data[0];

    return (
        <div className="col-4 mx-auto" >
            <h1 style={{"fontFamily": "Arial", "fontSize": "40px"}}> Frases de BreakingBad </h1>
            <hr />
            { 
                loading
                ?
                (
                    <div className="alert alert-info text-center">
                        Cargando...
                    </div>
                )
                :
                (
                    <>
                    <blockquote className="blockquote text-right">
                        <p>
                            { quote }
                        </p>
                        <footer className="blockquote-footer" >
                            { author }
                        </footer>
                    </blockquote>

                    <button 
                        className="btn btn-outline-info"
                        onClick= {() => increment()}
                        >
                        Siguiente
                    </button>

                    </>
                )
            
            }


        </div>
    )
}
