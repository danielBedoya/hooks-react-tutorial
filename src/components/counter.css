body{
    padding: 70px;
}

.complete {
    text-decoration: line-through;
}