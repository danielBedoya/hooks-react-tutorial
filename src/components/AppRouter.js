import React from 'react'
import { BrowserRouter as Router,
         Switch,
         Route} from "react-router-dom";
import { LoginScreen } from "./LoginScreen";
import { HomeScreen } from "./HomeScreen";
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <NavBar />
            <div>
                <Switch>
                    <Route exact path="/login" component= {LoginScreen} ></Route>
                    <Route exact path="/" component= {HomeScreen}></Route>
                </Switch>
            </div>
        </Router>
    )
}
