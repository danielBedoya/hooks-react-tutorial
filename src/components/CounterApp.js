import React from 'react'
import { useCounter } from "../hooks/useCounter";
import "./counter.css";

export const CounterApp = () => {

    const {state, increment, decrement, reset} = useCounter(20);

    return (
        <>
            <h1>Counter1 { state }</h1>

            <button
                className="btn btn-outline-primary" 
                onClick= { () => increment() }
                > +1 </button>
            
            <button
                className="m-2 btn btn-outline-primary" 
                onClick= { () => decrement() }
                > -1 </button>
            
            <button
                className="m-2 btn btn-outline-primary" 
                onClick= { reset }
                > Reset </button>

        </>
    )
}
