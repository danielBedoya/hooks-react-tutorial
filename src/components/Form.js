import React from 'react';
import { useForm } from "../hooks/useForm";

import './counter.css';

export const Form = () => {

    const [values, onHandleChange ] = useForm({
        name: '',
        email: '',
        password: ''
    });

    const {name, email, password} = values;

    return (
        <>
        <h1>Formulario Use Effect</h1>
        <form className="p-2">
            <input type="text" name='name' placeholder="Nombre" value={name} onChange={onHandleChange}></input>
            <hr />
            <input type="email" name='email' placeholder="Email" value={email} onChange={onHandleChange}></input>
            <hr />
            <input type="password" name='password' placeholder="ContraseÃ±a" value={password} onChange={onHandleChange}></input>
            <hr />
            <button type="submit" className="btn btn-outline-success" > Enviar </button>
        </form>
        </>
    );
}
