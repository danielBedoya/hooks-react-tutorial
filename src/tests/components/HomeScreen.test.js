import { mount } from 'enzyme';
import React from 'react';
import { UserContext } from "../../components/UserContext";
import { HomeScreen } from "../../components/HomeScreen";


describe("Pruebas en el componente HomeScreen", () => {
    const user = {
        name: "Daniel",
        email: "daniel@mail.com"
    }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    )

    test('Debe mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
})