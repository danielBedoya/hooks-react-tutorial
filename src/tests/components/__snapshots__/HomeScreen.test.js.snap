// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente HomeScreen Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  {
   "name": "Daniel",
   "email": "daniel@mail.com"
}
</div>
`;
